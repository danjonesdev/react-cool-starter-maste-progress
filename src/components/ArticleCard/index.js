import React from 'react';

import styles from './styles.scss';

type Props = { info: Object };

const ArticleCard = ({ info }: Props) => (
  <div className={styles.ArticleCard}>
    <h4>Article Card</h4>
    <ul>
      <li>Name: {info.name}</li>
      <li>Phone: {info.phone}</li>
      <li>Email: {info.email}</li>
      <li>Website: {info.website}</li>
    </ul>
  </div>
);

ArticleCard.defaultProps = {
  info: {
    name: '',
    phone: '',
    email: '',
    website: '',
  },
};

export default ArticleCard;
