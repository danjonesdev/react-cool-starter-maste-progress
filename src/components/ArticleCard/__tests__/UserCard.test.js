import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

import ArticleCard from '../index';

describe('<ArticleCard />', () => {
  test('renders', () => {
    const mockData = {
      name: 'Welly',
      phone: '007',
      email: 'test@gmail.com',
      website: 'www.test.com',
    };
    const tree = renderer.create(
      <StaticRouter location={''} context={{}}>
        <ArticleCard info={mockData} />
      </StaticRouter>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
