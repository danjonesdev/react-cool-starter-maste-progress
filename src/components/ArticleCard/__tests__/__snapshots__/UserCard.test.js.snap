// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> renders 1`] = `
<div
  className={undefined}
>
  <h4>
    Article Card
  </h4>
  <ul>
    <li>
      Name: 
      Welly
    </li>
    <li>
      Phone: 
      007
    </li>
    <li>
      Email: 
      test@gmail.com
    </li>
    <li>
      Website: 
      www.test.com
    </li>
  </ul>
</div>
`;
